src/repository/opamRepository.ml

```diff
@@ -396,10 +396,12 @@ let apply_repo_update repo = function
     Done ()
   | Update_err _ -> assert false
 
-let cleanup_repo_update = function
-  | Update_full d -> OpamFilename.rmdir d
-  | Update_patch f -> OpamFilename.remove f
-  | _ -> ()
+let cleanup_repo_update upd =
+  if not (OpamConsole.debug ()) then
+    match upd with
+    | Update_full d -> OpamFilename.rmdir d
+    | Update_patch f -> OpamFilename.remove f
+    | _ -> ()
```

Comment:
Indeed - but lots of things shouldn't generally happen :smile: I'm just thinking of terms of things which leave ~/.opam in a hard-to-recover state!

My solution with git apply should solve this problem very well - it falls back to patch if git isn't found (so we don't depend on git) and the most likely situation where this could happen is for patches generated when updating a git repo (because git will generate an LF-patch even for a CRLF repo) - but in that case the user must have git, or we couldn't have the git remote!